magento/pwa-studio
# #3185 fix - Duplicate react component key props

Fix for #3185

## Description

> Steps to reproduce the behavior:
> 
>     1. In a venia store with venia sample data, go to any product with colour variants
> 
>     2. Click on the default colour variant (matches the colour of the selected product's default carousel thumbnails).
> 
>     3. Click on an alternative colour, then click on the default colour again, repeat this
> 
>     4. Duplicate thumbnails for the default colour variant will be produced
> 
> 
> In the console react will warn that it has detected duplicate key props in the thumbnail list
> 
> **Expected behavior**
> 
> Should be no duplicate keys in the carousel's thumbnail list.

* carousel thumbnail keys are not currently unique based on item file + label

* item IDs are unique (within a products variants) so should be a good fix

## Changes

* Change ProductImageCarousel Thumbnail key to unique id

## Related Issue
Closes #3185


### Verification Steps
1. Go to any product page that has colour swatch variants.
2. Verify that selecting between the default variant and alternative variants does not produce duplicate react component keys.

## Screenshots / Screen Captures
> [here's a recording of this behaviour](https://i.imgur.com/GkpTSuo.mp4)
